about/about-ebpf.md

```diff
@@ -0,0 +1,160 @@
+| Kubernetes services       | kube-proxy iptables or IPVS mode        | BPF program and maps               |
+| IPIP                      | Supported                               | Supported (no performance advantage due to kernel limitations) |
+| VXLAN                     | Supported                               | Supported                          |
+| Wireguard                 | Supported                               | Supported                          | 
+| Other routing             | Supported                               | Supported                          | 
+| Supports third party CNI plugins | Yes (compatible plugins only)    | Yes (compatible plugins only)      |
+| Compatible with other iptables rules | Yes (can write rules above or below other rules) | Partial; iptables bypassed for workload traffic |
+| XDP DoS Protection        | Supported                               | Not supported (yet)                |
+| IPv6                      | Supported                               | Not supported (yet)                |
+| Host endpoint policy      | Supported                               | Not supported (yet)                |
+| Enterprise version        | Available                               | Not supported (yet)                |
+
+#### Architecture overview 
+
+{{site.prodname}}'s eBPF dataplane attaches eBPF programs to the `tc` hooks on each {{site.prodname}} interface as 
+well as your data and tunnel interfaces.  This allows {{site.prodname}} to spot workload packets early and handle them
+through a fast-path that bypasses iptables and other packet processing that the kernel would normally do.
+
+![Diagram showing the packet path for pod-to-pod networking; a BPF program is attached to the client pod's veth interface; it does a conntrack lookup in a BPF map, and forwards the packet to the second pod directly, bypassing iptables]({{site.baseurl}}/images/bpf-pod-to-pod.svg "Pod-to-pod packet path with eBPF enabled")
+
+The logic to implement load balancing and packet parsing is pre-compiled ahead of time and relies on a set of BPF 
+maps to store the NAT frontend and backend information.  The frontend map stores the metadata of the service, allowing
+for `externalTrafficPolicy` and "sticky" services to be honoured.  NodePorts are handled as a special case: after a
+miss in the service map, a second lookup is performed to check if the destination IP is local and whether the port is an
+active node port.
+
+![Detail of BPF program showing that conntrack happens first and a conntrack hit results in taking the fast path, possibly including applying NAT to the packet.  On conntrack miss, the frontend and backend NAT maps are consulted; after a hit, the program applies NAT to the packet. The packets then take a slower path that includes the kernel's RPF check,]({{site.baseurl}}/images/bpf-nat.svg "Expanded view of tc program with conntrack and NAT.")
```

Comment:
Pondering whether some of the diagrams are going into too much detail for an intro doc like this. Perhaps we could chat and explore that next week when we have a chance?

Reply:
I felt like they were getting a bit detailed but wasn't sure where to put the information if not there.